sympy/core/mul.py

```diff
@@ -1258,32 +1261,72 @@ def _eval_pos_neg(self, sign):
     def _eval_is_negative(self):
         if self.args[0] == -1:
             return (-self).is_positive  # remove -1
-        return self._eval_pos_neg(-1)
-
+        return self._eval_pos_neg(-1)     
+    
     def _eval_is_odd(self):
+ 
+  """ First we check to see if the expression as a whole is an integer because
```

Comment:
A comment would be better. It doesn't matter too much since this is a private method, but in general this sort of thing would go in a comment.